src/main/java/org/cactoos/proc/BiProcOf.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.proc;
+
+import org.cactoos.BiFunc;
+import org.cactoos.BiProc;
+import org.cactoos.Func;
+import org.cactoos.Proc;
+
+/**
+ * Represents many possible inputs as {@link BiProc}.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @param <X> Type of input
+ * @param <Y> Type of input
+ * @since 0.49
+ */
+public final class BiProcOf<X, Y> extends BiProcEnvelope<X, Y> {
+
+    /**
+     * Ctor.
+     * @param func The function
+     * @param <Z> Type of function output
+     */
+    public <Z> BiProcOf(final Func<X, Y> func) {
```

Comment:
@baudoliver7 in all the constructors, could you use proper covariance? The rules are explained in #1533

Reply:
@victornoel Covariance doesn't work for the last constructor. It works well for others constructors

Reply:
@baudoliver7 ah damn, why doesn't it work? Because the envelope itself has not the covariance? hmm, then let's forget about the last constructor then for now, we will resolve those question during #1533 :)

Reply:
@victornoel Done